base/lttemplates.dtx

```diff
@@ -351,12 +351,41 @@
 %   In the final argument of \cs{DeclareTemplateCode} the assignment of
 %   keys defined by the template may be delayed by including the command
 %   \cs{AssignTemplateKeys}. If this is \emph{not} present, keys are assigned
-%   immediately before the template code. If \cs{AssignTemplateKeys} is
+%   immediately before the template code. If an
+%    \cs{AssignTemplateKeys} command is
 %   present, assignment is delayed until this point. Note that the
 %   command must be \emph{directly} present in the code, not placed
 %   within a nested command/macro.
 % \end{function}
 %
+
+% \begin{function}{\SetTemplateKeys}
+%   \begin{syntax}
+%    \cs{SetTemplateKeys} \Arg{type} \Arg{template} \Arg{keyvals}
+%   \end{syntax}
+
+%    In the final argument of \cs{DeclareTemplateCode} one can
+%    also overwrite the current template key value settings by using the command
+%    \cs{SetTemplateKeys}, i.e., after the template default
+%    values and the values from the instance have been assigned.
+
+%    The main use case for this command is the situation where there
+%    is an argument (normally \texttt{\#1}) to to the template in which
+%    a key/value list can be specified that overwrites the normal
+%    settings. In that case one could use something like
+%\begin{quote}
+%  \verb/\tl_if_empty:nF {#1} { \SetTemplateKeys/\Arg{type}\Arg{template}\verb/{#1} }/
```

Comment:
Why the test here? You could just apply `\SetTemplateKeys` directly.

Reply:
yes one could, but